core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -494,7 +445,7 @@ public IndicesOptions getOptions() {
                 return expressions;
             }
 
-            if (expressions.isEmpty() || (expressions.size() == 1 && MetaData.ALL.equals(expressions.get(0)))) {
+            if (expressions.isEmpty() || (expressions.size() == 1 && (MetaData.ALL.equals(expressions.get(0))) || Regex.isMatchAllPattern(expressions.get(0)))) {
```

Comment:
In the case that just `*` is specified we can use the same shortcut as if `_all` or no expressions are specified.